### DISCUSSION TITLE
How to specify the deployment manifest file?

### DISCUSSION THREAD

**Discussion by Daxcor69** [argoproj/argo-cd#10228] [Q&A]
Hello I am struggling with getting argocd to read the application manifest file.  I have deployment.yaml in the root dir of the path and I also have package-lock.json.  Here is the error I am getting....
```Unable to create application: application spec for mainsite is invalid: InvalidSpecError: Unable to generate manifests in mainsite: rpc error: code = FailedPrecondition desc = Failed to unmarshal "package-lock.json": Object 'Kind' is missing in '{ "name": "website", ```

How do i tell Argo to use deployment.yaml for the manifest and not any other file?

Thanks

**Comment by kostis-codefresh** [comment]
Kubernetes manifests should not be in the same folder as source code. In fact, you should put Kubernetes manifests in a completely different Git repository than source code.

More details here https://codefresh.io/blog/argo-cd-best-practices/

**Reply by crenshaw-dev** [reply]
If you absolutely must use the same repo, you can use glob patterns to limit what Argo CD considers to be a manifest file. https://github.com/argoproj/argo-cd/blob/master/docs/operator-manual/application.yaml#L100-L105
